src/main/java/com/rapiddweller/benerator/factory/ComponentBuilderFactory.java

```diff
@@ -146,7 +146,8 @@ private static ComponentBuilder<?> createAlternativeGroupBuilder(
 
   private static ComponentBuilder<?> createPartBuilder(
       ComponentDescriptor part, Uniqueness ownerUniqueness, boolean iterationMode, BeneratorContext context) {
-    if (iterationMode) {
+    var containerValue = part.getDetailValue("container");
+    if (iterationMode && (containerValue == null || !containerValue.equals("list"))) {
```

Comment:
@dangsg Did you check out this article about parts ? What will happen with container="array" ?
https://docs.benerator.de/latest/distribution.html?h=container#weightednumbers